src/fibers_meta.jl

```diff
@@ -30,6 +30,43 @@ function fiber(arr, default=zero(eltype(arr)))
     Base.copyto!(Fiber((DenseLevel^(ndims(arr)))(Element{default}())), arr)
 end
 
+@generated function Base.:(==)(A::Fiber, B::Fiber)
```

Comment:
This looks great! Since the body of the generated function is the same for all three, perhaps we could add a helper function that all three overloads call? Something like `helper_equal(A, B)` that implements the generator. I should have done this for copyto! as well I think.